pkg/config/application.go

```diff
@@ -253,6 +255,14 @@ func (s *PipelineStage) UnmarshalJSON(data []byte) error {
 	s.Timeout = gs.Timeout
 
 	switch s.Name {
+	case model.StageCustomSync:
+		s.CustomSyncOptions = &CustomSyncOptions{}
+		if len(gs.With) > 0 {
+			err = json.Unmarshal(gs.With, s.CustomSyncOptions)
+		}
+		if s.CustomSyncOptions.Timeout <= 0 {
+			s.CustomSyncOptions.Timeout = defaultCustomSyncTimeout
+		}
```

Comment:
No need to do this, we set all default values here

https://github.com/pipe-cd/pipecd/blob/master/pkg/config/config.go#L215-L217

using https://github.com/creasty/defaults